### PR TITLE
Removes plasma man welding/flash protection

### PR DESCRIPTION
**Pull request by AarontheIdiot**
## About The Pull Request

 Gets rid of plasma man welding/flash protection.

## Why It's Good For The Game

 Nolan Basinger suggested I do this.

Plasma men are the typical race for powergamers to pick due to the fact that you get roundstart flash & welding protection, and it lasts all shift, unless you die of course. The plasma man race was never really supposed to be balanced, as it was initially added to the game in order to be a challenge. Free shift-start eye protection is not a challenge.
Plasmamen also have exceptional rad resistance and their helmets are space proof, meaning all they need to explore space is some extra plasma and a firesuit from the fire closets that scatter most stations. For a race intended to be challenging to play, this doesn't seem like much of a challenge. Plasmamen do also take 1.5x as much damage, but that seems almost too insignificant to be considered a challenge. 

## Changelog
:cl: AarontheIdiot
del: Plasma enviro helmets no longer have flash and welding protection.
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
You did this all wrong. The slowdown should be on the jumpsuit and they helmet should welding protection on the helmet should be a toggle.

**Comment by kriskog** [comment]
> 
> 
> You did this all wrong. The slowdown should be on the jumpsuit and the welding protection on the helmet should be a toggle.

This, but they dont actually need a slowdown.

**Comment by Fikou** [comment]
overshooting the mark here

**Comment by Ty-the-Smonk** [comment]
oh god rev plasmemes

**Comment by Qustinnus** [comment]
>speedmod 0.5

yeah this misses the mark hardcore. put the debuffs on the suit and remove the speed buff man. there's no point over-nerfing them.

**Comment by datorangebottle** [comment]
> This, but they dont actually need a slowdown.

Expanding on this, the slowdown was in initially and was removed because nobody liked it.

Just remove all the protections on the helmet. They can get a welding gas mask anyway, and it's still a flashlight that they constantly have on their person.

**Review by 81Denton — DISMISSED** [review]
The slowdown is unnecessary since plasmamen already take 150% burn/brute/heat damage. 

If you want to get rid of the the helmet's `flash_protect`, replace it with a welding helmet-style toggle. It could probably also use different sprites so other players can tell whether the visor is engaged or not.

**Comment by AarontheIdiot** [comment]
going to remove the slowdown very soon

**Comment by 81Denton** [comment]
Actually disregard what I said about the helmet, plasmamen can wear welding goggles as well as masks. No need for their helmets to be flashproof at all.

**Comment by duckay** [comment]
but how will you explain going blind with no eyes? how can mirrors be real if our eyes aren't real?

**Comment by Ziiro** [comment]
I've been playing a plasmeme on occasion recently and the flash protection did seem a little strange in the grand scheme of balance. 

**However**

The problem is I'm about 99.9% sure welding goggles can't be **created** anywhere. They're only available in the engievend, and the one in science.

IF you're going to move forward with this, you should probably add them to the autolathe.

**Comment by SynnGraffkin** [comment]
Don't plasmemes already have some kind of light toggle on their helmet? Just make that the welding protection with the reduced vision and done.

**Comment by datorangebottle** [comment]
> Don't plasmemes already have some kind of light toggle on their helmet? Just make that the welding protection with the reduced vision and done.

Or they could just wear gamer goggles /sunglasses underneath their helmet instead of starting with a welding helmet they get to take into permabrig with them.

**Comment by OmegaShoots** [comment]
Here's some things to consider.
Plasmamen cannot be defibbed, are hard(er) to clone properly, if they aren't recovered somewhat quickly after death, with the recent cremation addition, dying almost certainly means permanent round removal. They cannot at all survive fires, crit in like 3 laser shots, and are forced to use internals for the entire round.

Having roundstart flash protection is actually somewhat of a hinderance to them, because on paper this says they're harder to rev, in practice this means they're always systematically murdered in rev rounds except by the most benevolent of head revs who can robust and unhelmet them fast enough to get the flash off, and why go through that effort when they can during that entire process call for help.

They are not human and don't get asimov protection.

"But roundstart spaceproof flashproof helmet"

**Comment by MrFluffster** [comment]
> Here's some things to consider.
> Plasmamen cannot be defibbed, are hard(er) to clone properly, if they aren't recovered somewhat quickly after death, with the recent cremation addition, dying almost certainly means permanent round removal. They cannot at all survive fires, crit in like 3 laser shots, and are forced to use internals for the entire round.
> 
> Having roundstart flash protection is actually somewhat of a hinderance to them, because on paper this says they're harder to rev, in practice this means they're always systematically murdered in rev rounds except by the most benevolent of head revs who can robust and unhelmet them fast enough to get the flash off, and why go through that effort when they can during that entire process call for help.
> 
> They are not human and don't get asimov protection.
> 
> "But roundstart spaceproof flashproof helmet"

This man speaks the truth.

I think that making it a toggleable welding/flash protection is the best option, especially if the toggle means they get reduced visibility like with welding helmets while in use.
This slightly nerfs them but not in an unfun way, adds more gameplay(you either have your eyes protected or you can see normally), and stops them from being murder targets during rev rounds.

**Comment by datorangebottle** [comment]
> Plasmamen cannot be defibbed, are hard(er) to clone properly, if they aren't recovered somewhat quickly after death, with the recent cremation addition, dying almost certainly means permanent round removal. They cannot at all survive fires, crit in like 3 laser shots, and are forced to use internals for the entire round.

Being forced to use internals the entire round isn't a fucking downside in 99% of situations. You have an excuse to use internals the entire round and be immune to the majority of gas-based threats. You're immune to CO2, N2O, nitrogen, nonfire plasma, tritium. I'm sure there's some stuff I missed. 

You're immune to radiation, period, which enables some hilarious memes of being a traitor plasmeme engineer and running around with 25k rads and making everyone sick.

How often do you die and get your suit removed? Oh, when there's a fire or someone deliberately removes your suit to cremate you, which, there's lots of ways to remove someone from the round permanently already, and literally anybody can be cremated(or at least husked) with the new cremation changes. Nobody can reasonably deny you access to plasma(hey look another upside, free traitor objectives) unless you're proven to be an antag and then you can just break in and take it. 



> Having roundstart flash protection is actually somewhat of a hinderance to them

bruh.

you get to ignore a knockdown in a stun and knockdown based combat system, roundstart.
shut the fuck up.

> They are not human and don't get asimov protection.

Neither does any other alien race. 

The helmet is unjustifiably powerful and outweighs the downsides of taking plasmeme on its own, not to mention all the other things you get. Being able to spacewalk with an extremely common item is fucking stupid. A traitor plasmeme can, without any warning or need to grab a very limited item, practically roundstart, walk out and EMP grenade the AI sat until the AI is dead.

**Comment by datorangebottle** [comment]
Also, being set on fire without your suit is a mitigated downside, because it's useful as a weapon.

**Comment by GuppyLaxx** [comment]
you try using it as weapon
tip:you cant

mad r*tard who doesnt play plasmaman is mad because he got rev and couldnt flash a plasmaman

**Comment by OmegaShoots** [comment]
@datorangebottle I'm not sure exactly where all that hostility came from, but I think you need to take a step back and calm down?

**Comment by PKPenguin321** [comment]
What do you mean you can't? You totally can

**Comment by OmegaShoots** [comment]
> 
> 
> What do you mean you can't? You totally can

You can, 3 times, then you don't have any more self extinguishes and now have to rely on water/resisting, and if you get stunned you're fucked.

**Comment by datorangebottle** [comment]
> you try using it as weapon
> tip:you cant
> 
> mad r*tard who doesnt play plasmaman is mad because he got rev and couldnt flash a plasmaman

big thonk

i've played plenty of plasmaman, tritium XIV, Radium something, iron something
i don't remember the other two well because they were gimmicks, but tritium was a pacifist science main, radium was rad spam traitor engineer memes, and iron was a security officer


also i hate revs with a passion and i think being flash/weld immune roundstart is absolutely stupid




> @datorangebottle I'm not sure exactly where all that hostility came from, but I think you need to take a step back and calm down?

People treating upsides as downsides makes me unreasonably angry and I've had this argument a dozen times.

**Comment by PKPenguin321** [comment]
I guess? I've still seen plasmamen use it in fistfights and such though

**Comment by datorangebottle** [comment]
> > What do you mean you can't? You totally can
> 
> You can, 3 times, then you don't have any more self extinguishes and now have to rely on water/resisting, and if you get stunned you're fucked.

oh no my gamer move that causes people to waste time extinguishing themselves or take lots of extra damage is actually a slight risk

it's almost like it's fucking worth the risk if my opponent isn't countering that strategy by holding a fire extinguisher

**Comment by OmegaShoots** [comment]
@datorangebottle nobody else is being insulting or inflammatory, so why are you?
Can you post a reply that doesn't make it sound like you just came from /b/?

**Comment by datorangebottle** [comment]
> @datorangebottle nobody else is being insulting or inflammatory, so why are you?
> Can you post a reply that doesn't make it sound like you just came from /b/?

no you dumb fucking cretin, you fucking fool, absolute fucking buffoon, you bumbling idiot. fuck you.


respond to my actual points instead of complaining that i'm not being polite or don't bother at all, don't really care which

**Comment by SynnGraffkin** [comment]
Honestly, just making the welding portion of their helmet a toggle is a better change that flat out removal. Sure, welding gas masks exist but require research / access to science or engineering lathes, which most plasmamen are in those jobs anyways, but still something to point out. The only other welding protection would be a welding HELMET which can be printed at autolathes but removes your helmet, so that doesn't work. Sunglasses only work, to my knowledge, with the experimental welder, since other welders would blind you. The welding glasses are map spawn only it seems, and cannot be printed. So having the option to print welding glasses or helmets at an autolathe would be a decent change overall, not just for plasmemes.

**Comment by datorangebottle** [comment]
> Honestly, just making the welding portion of their helmet a toggle is a better change that flat out removal. Sure, welding gas masks exist but require research / access to science or engineering lathes, which most plasmamen are in those jobs anyways, but still something to point out. The only other welding protection would be a welding HELMET which can be printed at autolathes but removes your helmet, so that doesn't work. Sunglasses only work, to my knowledge, with the experimental welder, since other welders would blind you. The welding glasses are map spawn only it seems, and cannot be printed. So having the option to print welding glasses or helmets at an autolathe would be a decent change overall, not just for plasmemes.

being able to print welding goggles makes welding gas masks useless.
sunglasses severely lower the amount of eye damage you take from a welder.
having a roundstart unremovable welding helmet is not okay regardless, and every nonplasmaman has the issue of finding welding protection anyway! I don't see why we suddenly need to make welding goggles printable when this has been an issue everybody but plasmamen has had to deal with since the dawn of time.

**Comment by SynnGraffkin** [comment]
Welding gas masks wouldn't be useless though? You can still use them as your internals mask and welding combo if you so desired such.  Hell, some people may want them for fashion because it looks like a skull over your face, kind of like how I wear a welding helmet for fashion as a roboticist and would rather use goggles or gamer glasses.

**Comment by datorangebottle** [comment]
The only(very niche) use for a welding gas mask if goggles were that easy to get would be for having other glasses on. You start with an internals mask in your backpack.

**Comment by ExcessiveUseOfCobblestone** [comment]
@datorangebottle be mindful that your posts can be taken at face value by github mods.

Your EPIC / BASED arguments won't do much good if they get hidden because you're suspended, so probably should take his advice and tone down the hostility.

File: code/modules/clothing/spacesuits/plasmamen.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/spacesuits/plasmamen.dm]: double

File: code/modules/research/designs/autolathe_designs.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/designs/autolathe_designs.dm]: OOS for this PR imo, this can be circumvented by investing in Welding Gas Masks

**Comment by GuppyLaxx** [comment]
if you actually think you can use taking off your helmet as a plasmaman to fight you have zero experience using it and how it actually works
taking off your helmet generates a small amount of firestacks so miniscule that by the time you walk into them and put on your helmet the person already got put out
you're literally 10000000 times more efficient to do this by splashing them with welding fuel and hitting them with a lighter

**Comment by ExcessiveUseOfCobblestone** [comment]
Also in the future please don't touch species/species-specific interactions without consulting a maintainer prior to PRing or it can be closed on that reason alone.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Formalizing above

**Comment by SynnGraffkin** [comment]
And jobs that typically use eyewear that isn't welding goggles can still use welding gas masks if needed, @datorangebottle . 

Engineers with T-ray / Engie glasses or mesons. 
Scientists with science goggles. 
Medical with their huds. 
Security with their sechuds and wanting internals plus welding. 
Miners using NVGs.

**Comment by Dorsisdwarf** [comment]
Finally, geez

**Comment by datorangebottle** [comment]
> And jobs that typically use eyewear that isn't welding goggles can still use welding gas masks if needed, @datorangebottle .
> 
> Engineers with T-ray / Engie glasses or mesons.
> Scientists with science goggles.
> Medical with their huds.
> Security with their sechuds and wanting internals plus welding.
> Miners using NVGs.

Miners don't start with NVGs and normal NVGs aren't useful as a miner, they also don't really need to weld unless they have prosthetics.
Engineers only need to wear mesons in the engine room and you can use a t-ray scanner to achieve the same effect.
Science goggles are only really good for chemists, as far as i'm aware.
Security shouldn't be welding, but their huds count as sunglasses and have the reduced eye damage already. They also can't wear them with welding anything at the same time or they go completely blind.
Medical shouldn't be welding.
You forgot robotics and diagnostic huds but they start with helmets and goggles on most maps.

**Comment by 81Denton** [comment]
Imo just remove flash protection from the helmet and make welding goggles printable at the autolathe, the same way welding helmets are. If plasmemes need to weld something, they can just take a trip over to cargo.

**Comment by SynnGraffkin** [comment]
Or if they are an engineer, like most plasmemes, use their own autolathe engineering always makes.

**Comment by Narcissisko** [comment]
Bad idea. Have you ever used a welding tool on plasmamen? Probably not. Do it and find out why they should be immune to flashes.

**Comment by Detective-Google** [comment]
alien races are all meant to be overall negative relative to humans, giving people flash protection that is usually either rare (sunglasses) or negatively impacts you (welding helmets) roundstart is kinda goofy

It's really not a huge nerf to plasmemes to go grab some welding goggles and have to weld with actual protection like everyone else.

**Comment by wesoda25** [comment]
Tbh give plasmamen two options (in character setup): spaceproof enviro suit w/ slowdown, and normal enviro suit without. Keep flash protection.

**Comment by zeroisthebiggay** [comment]
here's an idea:
the helmet has three modes which are all cycleable
- the normal one with no downsides or upsides
- the one with a light on the front (it's a flashlight so maybe increase flashing vulnerability? doesn't make sense but balance wise it seems like a fair compromise to having a free flashlight glued to your head)
- and a third new one, a black welding shield folds over the visor of the helmet, providing full welding immunity but completely blinds the plasmaman, using it as a welding tool is fine since chances are you won't need to see, but using it as a combat tool (ie a cyborg entered the room and is _**very angry**_ with you) would result in you having to sacrifice vision or manning up and stealing a pair of sunglasses and or a welding gas mask.

**Comment by ExcessiveUseOfCobblestone** [comment]
FYI I told them to make a sep PR for the eyewear in autolathes and I'll merge this when I merge the other PR.

**Comment by dee-dubya** [comment]
Make the weld/flash protection into a toggle, as many many other people have already suggested. Give it the same vision restrictions that other welding protection does when it's toggled on, and the problem is solved.

**Comment by ExcessiveUseOfCobblestone** [comment]
it doesn't need intrinsic flash protection/welder protection though, it's a roundstart item.

**Comment by optimumtact** [comment]
I think deedubya's approach was better tbh

**Comment by lolman360** [comment]
> 
> 
> I think deedubya's approach was better tbh

then why'd you merge this, tutti-frutti?

**Comment by 81Denton** [comment]
I merged it, not oranges

**Comment by 81Denton** [comment]
Deedubya's idea is also good, but imo it should have separate helmet icon states so that other players can tell if the helmet currently has welding protection. If someone feels like spriting those I'll merge them.

**Comment by pireamaineach** [comment]
I mean this doesn't really make any sense. Plasmamen are always claimed to be a powergamer race but the 1.5x modifiers are always enough that powergamers simply don't play plasmamen. At this point you're just inconveniencing everyone because of a shitty meme claim that doesn't hold up against scrutiny.

**Pull request merged**
